cmake/opentelemetry-build-external-component.cmake

```diff
@@ -13,13 +23,18 @@
 if(OPENTELEMETRY_EXTERNAL_COMPONENT_PATH)
   # Add custom component path to build tree and consolidate binary artifacts in
   # current project binary output directory.
-  add_subdirectory(${OPENTELEMETRY_EXTERNAL_COMPONENT_PATH}
-                   ${PROJECT_BINARY_DIR}/external)
+  foreach(DIR IN LISTS OPENTELEMETRY_EXTERNAL_COMPONENT_PATH)
+    get_directory_name_in_path(${DIR} EXTERNAL_EXPORTER_DIR_NAME)
+    add_subdirectory(${DIR} ${PROJECT_BINARY_DIR}/external/${EXTERNAL_EXPORTER_DIR_NAME})
```

Comment:
nit - can there be a conflict  if the path list is (say) `/dir1/subdir1 /dir2/subdir1`.

Reply:
cmake use `;` to split list item, not space.I think there will no conflict here.